### ISSUE TITLE
Cannot receive push noti in android

### ISSUE THREAD

**Issue by Thaina** [OneSignal/OneSignal-Unity-SDK#310]
**Environment**

1. What version of the Unity OneSignal SDK are you using?

* 2.13.0

1. Provide a list of your project dependencies

* Firebase, Facebook, VoxelBusters

1. Does the issue occur only in iOS, Android, etc?

* Only android

Build from unity 2020.2a15

**Steps to Reproduce Issue:**

1. Import the latest OneSignalSDK.unitypackage into project
1. Open app and init
1. Attempt to receive a push notification, sending test noti from dashboard

This below is onesignal log I got

```
07-30 17:12:32.871 23240 23683 I OneSignal: Restoring notifications
07-30 17:12:32.872 23240 23683 I OneSignal: Querying DB for notifs to restore: created_time > 1595499152 AND dismissed = 0 AND opened = 0 AND is_summary = 0 AND expire_time > 1596103952
07-30 17:12:32.878 23240 23309 I OneSignal: Location permissions not added on AndroidManifest file
07-30 17:12:33.099 23240 23701 I OneSignal: Device registered, push token = cG5bbhVbRASB9... something something
07-30 17:12:39.005 23240 23890 I OneSignal: Device registered, UserId = xxxx
07-30 17:14:21.242 23240 23682 I OneSignal: scheduleSyncServiceAsJob:result: 1
07-30 17:15:03.871 23240 24205 I OneSignal: Location permissions not added on AndroidManifest file
07-30 17:15:03.872 23240 24205 E OneSignal: Error updating the user record because of the null user id
```

**Comment by rgomezp** [comment]
Note: this has been [resolved](https://github.com/OneSignal/OneSignal-Android-SDK/issues/1008) and will be included in the next Unity release.

**Comment by Thaina** [comment]
@rgomezp Is that will solved the problem of not receive push noti?

**Comment by jkasten2** [comment]
@Thaina It looks like your device is being subscribed for push based on this line in the log
```
07-30 17:12:33.099 23240 23701 I OneSignal: Device registered, push token = cG5bbhVbRASB9...
```
Can you confirm you can find your device on the OneSignal dashboard under Audience > All Users?
If the notification shows as successful from the OneSignal dashboard when you send one please see this troubleshooting guide.
https://documentation.onesignal.com/docs/notifications-show-successful-but-are-not-being-shown

**Comment by Thaina** [comment]
@jkasten2 I have try all those and none was success

I can even send test message from firebase console and it was shown noti properly. But onesignal message sending in the same time are not

**Comment by jkasten2** [comment]
@Thaina Can you confirm the "LAST ACTIVE" field time is being updated on app.onesignal.com under  Audience > All Users for your specific device? This will ensure the device is still able to send updates to OneSignal and you are sending your test notification to the correct device.

If the above is correct I recommend giving your device a few minutes to ensure it is not having any networking issues. Then search the logcat for your package name to ensure your app wasn't killed for any reason if it still was not received.

**Comment by Thaina** [comment]
@jkasten2 Just testing it last hour

![image](https://user-images.githubusercontent.com/1042507/89385458-daa37d80-d729-11ea-9f46-4da05bcaadb3.png)

![image](https://user-images.githubusercontent.com/1042507/89385522-f0b13e00-d729-11ea-877e-c1706ae111ad.png)

**Comment by Thaina** [comment]
@jkasten2 I have said so many many times that I could send firebase message to let it show noti from firebase test console like this

![image](https://user-images.githubusercontent.com/1042507/89385676-2fdf8f00-d72a-11ea-8691-e3e7b66d7c4c.png)

It always being shown properly from the start to even just a few second before I reply this comment

Unlike this onesignal test console that never send anything to me for such times

![image](https://user-images.githubusercontent.com/1042507/89385859-77feb180-d72a-11ea-9c4d-743bf57c1c39.png)

I always trying to send it at the same time. Only firebase message got noti. Onesignal never appear

### So shouldn't you stop thinking about networking or app killed already????

**Comment by Thaina** [comment]
@jkasten2 Additionally

I can also confirm that, OneSignal has registered push token with the same token from firebase and device

I have check the push token column in Audience console, that row in the capture image, has the exact same push token with device. And so I can use that token in firebase test message button

**Comment by askanhan** [comment]
Exactly same problem here

**Comment by leszekkrol** [comment]
Same issue

**Comment by nagibator96** [comment]
Same

**Comment by jkasten2** [comment]
A fix for this is now available in [2.13.3 release](https://github.com/OneSignal/OneSignal-Unity-SDK/releases/tag/2.13.3)! You can safely remove the workaround noted here.
